ampd/src/monitoring/endpoints/metrics.rs

```diff
@@ -433,3 +564,22 @@ mod tests {
         result.join("\n") + "\n"
     }
 }
+
+/// System metrics like CPU and memory usage are inherently dynamic and change between test runs
+/// This function normalizes the system metrics to 0 for consistent output in golden file tests.
+#[cfg(test)]
+pub mod test_utils {
+    pub fn normalize_system_metrics(text: &str) -> String {
+        let mut result = Vec::new();
+        for line in text.lines() {
+            if line.starts_with("ampd_cpu_usage_percent") && !line.starts_with("#") {
+                result.push("ampd_cpu_usage_percent 0".to_string());
+            } else if line.starts_with("ampd_memory_usage_bytes") && !line.starts_with("#") {
+                result.push("ampd_memory_usage_bytes 0".to_string());
+            } else {
+                result.push(line.to_string());
+            }
+        }
```

Comment:
this completely overwrites the line, which won't show any discrepancies if something about the output changes. If my assumption that the last part of the line can be a float (or integer) is correct, I'd rather only replace that:

```suggestion
        let trailing_number = Regex::new(r"\d+(\.\d+)?$").unwrap();
        for line in text.lines() {
            if line.starts_with("ampd_cpu_usage_percent") || line.starts_with("ampd_memory_usage_bytes") {
                result.push(trailing_numer.replace(line, "BLANKED_FOR_TEST"));
            } else {
                result.push(line.to_string());
            }
        }
```
you don't need to check for `#` because it's incompatible with your other `starts_with` checks